### PR TITLE
[shared_preferences] Make a clone of mutable containers before set/get.

### PR DESCRIPTION
**Pull request by devxpy**
Fixes https://github.com/flutter/flutter/issues/28015

### CODE REVIEW

File: packages/shared_preferences/lib/shared_preferences.dart [hunk 1]
```diff
@@ -74,9 +80,11 @@ class SharedPreferences {
   /// exception if it's not a string set.
   List<String> getStringList(String key) {
     List<Object> list = _preferenceCache[key];
-    if (list != null && list is! List<String>) {
-      list = list.cast<String>().toList();
-      _preferenceCache[key] = list;
+    if (list != null) {
+      list = List<String>.from(list);
+      if (list is! List<String>) {
```
Comment (apaatsio) [inline packages/shared_preferences/lib/shared_preferences.dart]: Can this `if` condition ever be true? On line 84, we assign `list = List<String>.from(list)`, so is it even possible for `list` to be anything else than `List<String>` on line 85?

**Comment by neelu004** [comment]
Hi, I am adding the dependency of shared_preference and it is working fine when I am running my project on android using windows but when I am using Mac(Xcode) it is giving me error as ld: warning: directory not found for option '-L/Users/ghanshyam/Development/Android/Workspace/my_app/build/ios/Debug-iphoneos/shared_preferences'
ld: library not found for -lshared_preferences
clang: error: linker command failed with exit code 1 (use -v to see invocation)
I had tried using the following methods but that doesn't works for me. They are:-
pod install
pod upgrade
flutter clean
flutter packages get
flutter packages upgrade
flutter run using command it is working fine but works on simulator still could not run on my real iPhone device using the same command.
also when I am using flutter build ios, this will creates the Runner.app but how can I install this Runner.app on my iPhone device.
Please suggest some solution for the same.
Thanks!

File: packages/shared_preferences/lib/shared_preferences.dart (hunk 1 above)
Comment (devxpy) [reply packages/shared_preferences/lib/shared_preferences.dart]: Looks like an overlook on my part. Sorry! Will issue a fix.

Comment (devxpy) [reply packages/shared_preferences/lib/shared_preferences.dart]: @apaatsio Does it look better now?

**Comment by stuartmorgan** [comment]
This was addressed in https://github.com/flutter/plugins/pull/1276, so closing.

We apologize for the long delay in triaging this PR. We’re in the process of overhauling our PR triage system to respond much more quickly, as well as working through the backlog.

**Pull request closed without merging**